pkg/storage/bloom/v1/ast_extractor.go

```diff
@@ -89,11 +109,163 @@ func buildLabelMatcher(filter log.LabelFilterer) LabelMatcher {
 	}
 }
 
+// buildSimplifiedRegexMatcher builds a simplified label matcher from a regex.
+// reg may be mutated.
+func buildSimplifiedRegexMatcher(key string, reg *regexsyn.Regexp) LabelMatcher {
+	switch reg.Op {
+	case regexsyn.OpAlternate:
+		util.ClearCapture(reg)
+
+		left := buildSimplifiedRegexMatcher(key, reg.Sub[0])
+		for _, sub := range reg.Sub[1:] {
```

Comment:
I guess `len(reg.Sub)` is always >= 2, otherwise the regex is not valid.

Reply:
I can add in a length check to be safe, but I don't think it's possible to have `len(reg.Sub)==1` here. Even `warn|` will have two subexpressions, where the latter is an empty string match.